lib/view_model/settings/tor_view_model.dart

```diff
@@ -0,0 +1,117 @@
+  TorViewModelBase(this._settingsStore) {
+    reaction((_) => torConnectionMode, (TorConnectionMode mode) async {
+      if (mode == TorConnectionMode.enabled || mode == TorConnectionMode.torOnly) {
+        startTor();
+      } else {
+        stopTor();
+      }
+    });
+  }
+
+  bool torStarted = false;
+  final SettingsStore _settingsStore;
+  Tor torInstance = Tor.instance;
+
+  @action
+  Future<void> updateStartOnLaunch(bool value) async {
+    _settingsStore.shouldStartTorOnLaunch = value;
+  }
+
+  @computed
+  TorConnectionMode get torConnectionMode => _settingsStore.torConnectionMode;
+
+  @observable
+  TorConnectionStatus torConnectionStatus = TorConnectionStatus.disconnected;
+
+  @action
+  void setTorConnectionMode(TorConnectionMode mode) => _settingsStore.torConnectionMode = mode;
+
+  Future<void> connectOrDisconnectNodeToProxy({required bool connect}) async {
+    final appStore = getIt.get<AppStore>();
+    if (appStore.wallet != null) {
+      final node = _settingsStore.getCurrentNode(appStore.wallet!.type);
+      if (connect && (node.socksProxyAddress?.isEmpty ?? true)) {
+        node.socksProxyAddress = "${InternetAddress.loopbackIPv4.address}:${torInstance.port}";
+      } else if (!connect) {
+        node.socksProxyAddress = null;
+      }
+
+      bool torOnly = _settingsStore.torConnectionMode == TorConnectionMode.torOnly;
+      if ([WalletType.bitcoin, WalletType.litecoin, WalletType.bitcoinCash]
+          .contains(appStore.wallet!.type)) {
+        bitcoin!.setTorOnly(appStore.wallet!, torOnly);
+      }
+
+      await appStore.wallet!.connectToNode(node: node);
+    }
+  }
+
+  @action
+  Future<void> startTor() async {
+    try {
+      torConnectionStatus = TorConnectionStatus.connecting;
+
+      // stop monero from syncing before tor is connected by connecting to a dummy node:
+      final appStore = getIt.get<AppStore>();
+      if (appStore.wallet != null && appStore.wallet!.type == WalletType.monero) {
+        appStore.wallet!.syncStatus = NotConnectedSyncStatus();
+        await appStore.wallet!.connectToNode(node: Node(uri: "http://127.0.0.1"));
+      }
+
+      if (!torStarted) {
+        torStarted = true;
+        torInstance = await Tor.init();
+      }
+
+      await torInstance.enable();
+
+      _settingsStore.shouldStartTorOnLaunch = true;
+
+      SocksTCPClient.setProxy(proxies: [
+        ProxySettings(
+          InternetAddress.loopbackIPv4,
+          torInstance.port,
+          password: null,
+        ),
+      ]);
+
+      torConnectionStatus = TorConnectionStatus.connected;
+
+      // connect to node through the proxy:
+      await connectOrDisconnectNodeToProxy(connect: true);
+    } catch (e) {
+      torConnectionStatus = TorConnectionStatus.disconnected;
+    }
+  }
+
+  @action
+  Future<void> stopTor() async {
+    torInstance.disable();
+    // setting the torConnectionMode to disabled will prevent anything from actually using the proxy
```

Comment:
isn't that what we want?
if so, then I believe you should set the torConnectionMode to disabled

Reply:
I think this comment is just poorly worded, this function is triggered whenever the user sets torConnectionMod to disabled

Setting it again in this function would (I think) cause an infinite loop